GUI/Main.cs

```diff
@@ -1200,6 +1206,73 @@ private void reportAnIssueToolStripMenuItem_Click(object sender, EventArgs e)
         {
             Process.Start("https://github.com/KSP-CKAN/NetKAN/issues/new");
         }
+
+        private void ModList_MouseDown(object sender, MouseEventArgs e)
+        {
+            var rowIndex = ModList.HitTest(e.X, e.Y).RowIndex;
+
+            // Ignore header column to prevent errors
+            if (rowIndex != -1 && e.Button == MouseButtons.Right)
+            {
+                // Detect the clicked cell and select the row
+                ModList.ClearSelection();
+                ModList.Rows[rowIndex].Selected = true;
+
+                // Show the context menu
+                ModListContextMenuStrip.Show(ModList, new Point(e.X, e.Y));
+
+                // Set the menu options
+                var guiMod = (GUIMod)ModList.Rows[rowIndex].Tag;
+
+                downloadContentsToolStripMenuItem.Enabled = !guiMod.IsCached;
+
+                if (guiMod.IsInstalled || guiMod.IsAutodetected)
```

Comment:
Properly handling AD mods is a whole can of worms, that I think would be better handled in a separate PR. There's a lot of different cases to consider. As this is, this throws a ModNotInstalledKraken for AD mods in ModuleInstaller.UninstallList that isn't being caught in the transaction below and thus crashes the client. Can we just make this `if ( guiMod.IsInstalled )`? Then I can merge this and we can all work on extending the right-click functionality one menu item at a time.